src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1797,6 +1815,56 @@ export abstract class AbstractConfigureRecommendedExtensionsAction extends Actio
 			}));
 	}
 
+	protected addRecommendedExtensionToFolder(extensionsFileResource: URI, extensionId: string): TPromise<any> {
+		return this.getOrCreateExtensionsFile(extensionsFileResource)
+			.then(({ content }) => {
+				const jsonContent: IExtensionsConfigContent = json.parse(content) || {};
+				const folderRecommendations = jsonContent.recommendations || [];
+
+				if (folderRecommendations.map(e => e.toLowerCase()).indexOf(extensionId.toLowerCase()) !== -1) {
+					return TPromise.as(null);
+				}
+				folderRecommendations.push(extensionId);
+
+				const folderUnwantedRecommedations = jsonContent.unwantedRecommendations || [];
+				let index = -1;
+				for (let i = 0; i < folderUnwantedRecommedations.length; i++) {
+					if (folderUnwantedRecommedations[i].toLowerCase() === extensionId.toLowerCase()) {
+						index = i;
+						break;
+					}
+				}
```

Comment:
I would do just one `.toLowerCase()` at the top just to not have to recalculate it again in each iteration. Realistically speaking pretty insignificant given the small size of the arrays and strings though

Reply:
done